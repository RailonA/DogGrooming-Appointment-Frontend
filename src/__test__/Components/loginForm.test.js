import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import LoginForm from '../../Components/loginForm';

describe('LoginForm', () => {
  const mockStore = configureStore([]);
  const store = mockStore({});
  store.dispatch = jest.fn();

  it('should match the snapshot', () => {
    const wrapper = renderer.create(
      <Provider store={store}>
        <LoginForm onCancel={jest.fn} />
        <form>
          <div className="modal-header">
            <div
              className="modal-title h4"
            >
              Sign Up
            </div>
          </div>
          <div
            className="modal-body"
          >
            <div
              className="flex-column"
            >
              <input
                className="d-flex col-10"
                name="username"
                onChange={[Function]}
                placeholder="Enter Username"
                type="text"
              />
              <input
                className="d-flex col-10"
                name="password"
                onChange={[Function]}
                placeholder="Enter Password"
                type="password"
              />
              <input
                className="d-flex col-10"
                name="password_confirmation"
                onChange={[Function]}
                placeholder="Confirm Password"
                type="password"
              />
            </div>
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              type="submit"
            >
              SIGN UP
            </button>
            <button
              className="btn btn-secondary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Cancel
            </button>
          </div>
        </form>
      </Provider>,
    ).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
