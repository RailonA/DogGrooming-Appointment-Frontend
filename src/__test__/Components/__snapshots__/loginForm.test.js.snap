// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match the snapshot 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <div
      className="modal-header"
    >
      <div
        className="modal-title h4"
      >
        LOG IN
      </div>
      <button
        aria-label="Close"
        className="btn-close"
        onClick={[Function]}
        type="button"
      />
    </div>
    <div
      className="modal-body"
    >
      <input
        name="username"
        onChange={[Function]}
        placeholder="Enter Username"
        type="text"
      />
      <input
        name="password"
        onChange={[Function]}
        placeholder="*****"
        type="password"
      />
    </div>
    <div
      className="modal-footer"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        type="submit"
      >
        Log In
      </button>
      <button
        className="btn btn-secondary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Close
      </button>
    </div>
  </form>,
  <form>
    <div
      className="modal-header"
    >
      <div
        className="modal-title h4"
      >
        Sign Up
      </div>
    </div>
    <div
      className="modal-body"
    >
      <div
        className="flex-column"
      >
        <input
          className="d-flex col-10"
          name="username"
          onChange={
            Array [
              [Function],
            ]
          }
          placeholder="Enter Username"
          type="text"
        />
        <input
          className="d-flex col-10"
          name="password"
          onChange={
            Array [
              [Function],
            ]
          }
          placeholder="Enter Password"
          type="password"
        />
        <input
          className="d-flex col-10"
          name="password_confirmation"
          onChange={
            Array [
              [Function],
            ]
          }
          placeholder="Confirm Password"
          type="password"
        />
      </div>
    </div>
    <div
      className="modal-footer"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        type="submit"
      >
        SIGN UP
      </button>
      <button
        className="btn btn-secondary"
        disabled={false}
        onClick={
          Array [
            [Function],
          ]
        }
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>,
]
`;
